### PR TITLE
Adds semantic versioning and release publishing

### PR DESCRIPTION
**Pull request by AlexandruAntonescuKeysight**
## Summary
This PR is intended to enhance the CI pipeline and simplify the modules calling. It has the following additions:

A new workflow file was created for versioning task and it contains the following:
1. installs protolock for backward compatibility checking
2. adds semantic versioning and backward compatibility checking

- a few files have been added: VERSION, version.py, proto.lock
  - **_VERSION_** contains the version (major.minor.patch) in plain text; it is necessary for manually updating the major and dynamic version retrieval in pyproject.toml when installing with pip 
  - **_proto.lock_** is a file generated by running the command `protolock init` (only once, after that when there are modification it is ran `protolock commit`); this tool is used for checking protobuff backward compatibility
  - **_version.py_** is a python script that checks if backward compatibility is broken by calling `protolock status` and if so, the minor version is incremented, otherwise the patch number is incremented. Also, it manages the incrementing step of the version  and also updates the VERSION file with the new version

- the pyproject.toml was also updated to dynamically read the version from the VERSION file when building or installing
- also after the version is updated, a commit is made for submitting the new VERSION and proto.lock files to the repo
- the previous commit is done by an action and the commiter is `actions-user`. Because of this, `actions-user` was added to the allow list in the CLA-bot workflow

4. adds steps for publishing the release on the main project page under the **_Releases_** section
5. adds steps for building a python wheel of the project and publishing it to the assets section of the last added release

## Test Plan
I've tested this by locally running version.py script after randomly making changes to the proto (to check if protolock is working fine).
Also, I've made test branches on my fork and on Dan's to see if the workflows are working good on github runners.

### CODE REVIEW

**Comment by TaekyungHeo** [comment]
cc: @srinivas212
This PR has been thoroughly tested by Dan and @AlexandruAntonescuKeysight. It automatically increases the minor version of semantic versioning when there is a minor change in the protobuf definition. The major version should be updated manually.

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ __pycache__/
 *.et
 *.dot
 .pyre
+.vscode/
```
Comment (briancoutinho) [inline .gitignore:10]: nit seems a bit user editor specific

**Comment by briancoutinho** [comment]
Looks good (y)
QQ @AlexandruAntonescuKeysight  what happens when we increment major version, then the code need to not be compatible with previous minor, it was not clear how to do that with version.py

File: .gitignore (hunk 1 above)
Comment (AlexandruAntonescuKeysight) [reply .gitignore]: I've removed it

**Review by srinivas212 — APPROVED** [review]
Reviewed offline. Thank you for your contributions @AlexandruAntonescuKeysight!

**Pull request merged**